.github/actions/linux-build/action.yml

```diff
@@ -0,0 +1,73 @@
+name: "Linux Build"
+description: "Build the C++ library on Linux."
+inputs:
+  config: 
+    description: "Configuration to build (Release or Debug, see CMAKE_BUILD_TYPE)"
+    required: true
+    default: "Release"
+  compiler:
+    description: "Compiler to use (clang or gcc)."
+    required: true
+    default: "gcc"
+  version:
+    description: "Version of the compiler to use (min or latest)."
+    required: true
+    default: "latest"
+  coverage:
+    description: "Turn coverage on (ON or OFF)"
+    required: false
+    default: "OFF"
+  sanitize-ub:
+    description: "Turn on UB sanitzer (ON or OFF)"
+    required: false
+    default: "OFF"
+  sanitize-addr:
+    description: "Turn on address sanitzer (ON or OFF)"
+    required: false
+    default: "OFF"
+runs:
+  using: "composite"
+  steps:
+    - name: Install dependencies
+      shell: bash
+      run: |
+        sudo apt-get -qq update
+        sudo apt-get -qq -y install libhdf5-dev lcov
+    - name: Build
+      shell: bash
+      run: |
+        cd cpp
+        if [[ "${{ inputs.compiler }}" == "gcc" ]]; then
+          if [[ "${{ inputs.version }}" == "min" ]]; then
+            export CC=/usr/bin/gcc-7
+            export CXX=/usr/bin/g++-7
+          else        
+            export CC=/usr/bin/gcc-10
+            export CXX=/usr/bin/g++-10
+          fi
+        elif [[ "${{ inputs.compiler }}" == "clang" ]]; then
+          if [[ "${{ inputs.version }}" == "min" ]]; then
+            export CC=/usr/bin/clang-9
```

Comment:
I have no experience on the distribution of clang compiler versions but is 9 as a minimum not rather too new? Or is there any reason we need at least version 9 to get our code compiled?

Reply:
> I will have a more detailed look if you are finished :)

I consider it finished (or reviewable), some issues remain but I don't want to make this issue bigger.

Reply:
clang users tend to be progressive, in my experience. but the reason is that 9 is the lowest version installed on any of the github runners. 

Ubuntu 18.04 has clang 6 as a default version in apt, maybe that's a good choice? It should be able to build c++14 according to cppreference.

Reply:
created issue #70 to address this.